### ISSUE TITLE
Move to WHATWG

### ISSUE THREAD

**Issue by annevk** [whatwg/webidl#1016]
I've started thinking about the steps needed to move this document to the WHATWG and I would appreciate some review (feel free to edit this comment as needed):

- [x] Create `webidl.spec.whatwg.org` and ensure it can be deployed to. @foolip?
- [x] Create a logo. Anyone ideas? https://github.com/whatwg/whatwg.org/pull/377
- [x] Ensure the Twitter account is associated with the Twitter bot.
- [x] Create a PR that updates `index.bs` to follow WHATWG conventions. See #1018.
- [x] Align wpt directory name with shortname: https://github.com/web-platform-tests/wpt/pull/30327.
- [x] @heycam to move the repository to the WHATWG GitHub organization.
- [x] @heycam to set up a redirect for `heycam.github.io/webidl/`. Presumably as per https://gist.github.com/domenic/1f286d415559b56d725bee51a62c24a7.
- [x] Update the various repository settings, such as URL, description, branches, etc.
- [x] Update the labels.
- [x] Rename the master branch to main.
- [x] Merge https://github.com/whatwg/sg/pull/167. (This should probably not happen too soon as some tooling uses this database.)
- [x] Merge the PR for `index.bs`.
- [x] Run https://github.com/whatwg/spec-factory and merge the resulting PR. (We wanted to update spec-factory as well for this if I recall correctly. We could decide to not block on that and instead modify the resulting PR as needed to keep some of the existing modifications.) See https://github.com/whatwg/spec-factory/pull/31. The output is already included in the `index.bs` PR.
- [x] Check the box in https://github.com/w3c/whatwg-coord/issues/3.
- [x] Update wpt readme to point to the spec: https://github.com/web-platform-tests/wpt/pull/30355.
- [x] Ensure participate.whatwg.org is updated: https://github.com/whatwg/sg/issues/171
- [x] Make changes to specref https://github.com/tobie/specref/pull/686
- [x] Changes to Shepherd?
- [x] Publish a RD draft at the appropriate time (the first one needs some help)

cc @heycam @TimothyGu @domenic @EdgarChen @foolip

**Comment by domenic** [comment]
> Create a logo. Anyone ideas?

Curly brackets, e.g. `{}`, perhaps? Not sure the font is great but here's a hacked-together idea of what it would look like: https://jsbin.com/jefomakuma/edit?html,output

**Comment by bathos** [comment]
There's probably not gonna be anything iconographic that communicates it better so I think braces work. Though if it were `{};` I'd know _exactly_ what language it is given how often I forget they're required after blocks in Web IDL :)

This is probably just a local font thing, but what rendered for me is missing a space between the braces. Hopefully the real thing would have a gap; without, it this kinda reads as "WHATWG With Wings":

![screenshot of icon preview showing the two braces are flush, creating what looks more like a maxipad icon than a "code" icon](https://user-images.githubusercontent.com/6257356/131396314-75b043c7-3896-45a4-908f-36838c65e773.png)

**Comment by domenic** [comment]
Haha that's a funny rendering. FWIW here's what I saw. But in the end we need to make it an actual `<path>`, and probably pick a more "code-like" font.

![image](https://user-images.githubusercontent.com/617481/131401254-66b85609-2056-4ee7-adef-56cb0f3ede32.png)

**Comment by domenic** [comment]
https://fonts.google.com/?category=Monospace&preview.text=%7B%7D&preview.text_type=custom is probably the place to search for fonts

**Comment by annevk** [comment]
Hah, I was also thinking about curly brackets after posting this! Though I thought that perhaps it should be just `{` to align with `?`, `!`,  and `�`.

**Comment by foolip** [comment]
For webidl.spec.whatwg.org, I intend to try hosting it on AWS S3, as a first step to hosting all of our specs and static sites somewhere other than the current marquee VM, see https://github.com/whatwg/misc-server/issues/107.

**Comment by annevk** [comment]
What do you all think about
```xml
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 825 825">
  <circle cx="412.5" cy="412.5" r="371.25" fill="#fff" stroke="#3c790a" stroke-width="82.5"/>
  <path d="M249.10714 387.55814v48.92442c22.50001 0 40.17858 4.79651 53.03572 12.47093 14.46429 7.67442 22.49999 19.18605 27.32143 32.61628 4.82141 13.43024 6.42857 37.41279 6.42857 70.98837 0 33.57559 1.60714 55.63954 3.21429 66.19186 3.21429 17.26745 8.03571 30.69767 14.46429 40.2907 6.42856 10.55232 16.07141 18.22675 25.71427 23.98256 9.64285 5.75581 24.10714 10.55232 38.57145 14.38954 11.24999 1.9186 28.92856 3.8372 53.0357 3.8372H495v-47.00581h-12.85714c-28.92857 0-49.82142-3.83721-59.4643-11.51162-9.64285-7.67443-14.46426-23.98257-14.46426-50.84303 0-49.88372-1.60716-81.5407-3.2143-94.97094-4.82142-22.06395-12.85714-40.29069-27.32143-53.72092-14.46427-13.43024-33.75-23.98256-57.85713-30.69768 32.14285-10.55232 56.25-25.90116 69.10712-44.12791 12.85716-18.22674 19.28574-48.92442 19.28574-92.09302 0-39.33139 0-62.35465 1.60714-70.02907 1.60713-13.43023 8.03572-23.02326 16.07141-27.81977 9.64287-4.79651 27.32144-7.67442 54.64286-7.67442h12.85715V123.75h-24.10715c-27.32142 0-48.21427 1.91861-61.07141 4.79651-17.67858 4.79651-33.75001 11.51163-45 22.06396-11.25001 10.55232-19.28573 23.98256-22.50001 39.33139-3.21429 16.30814-4.82143 42.2093-6.42859 77.70349 0 36.45349-3.21427 61.39535-6.42855 74.82558-4.82144 13.43023-14.4643 24.94186-27.32144 32.61628-14.46427 7.67442-32.14286 12.47093-53.03571 12.47093z" fill="#3c790a"/>
</svg>
```
?

**Comment by bathos** [comment]
i love it, weight is more consistent with the circle-question-mark icon and it is just _slightly_ psychologically unnerving to see an unmatched brace (good)

**Comment by domenic** [comment]
LGTM!

**Comment by TimothyGu** [comment]
@annevk Looks good!

(Here's a render for convenience)

![webidl-whatwg](https://user-images.githubusercontent.com/1538624/132056645-24829533-19b3-41e4-830e-b954597b7f7b.png)

**Comment by annevk** [comment]
Status: at this point I consider everything ready for the move. What remains:

- @foolip to create the space. (This has to happen first.)
- @domenic to update participate.whatwg.org with `@webidl` Twitter credentials. (This matters less, but I suspect it'll happen soonish as it's trivial.)
- @heycam to move the repository. (This should happen once @foolip is done and ideally when several people are around to complete the remaining tasks.)

Sound good?

**Comment by foolip** [comment]
I'm trying to make this the first static site we host on S3, but if it proves too time-consuming to figure out I will fall back to putting it on marquee.

**Comment by foolip** [comment]
https://webidl.spec.whatwg.org/ exists now, I've checked some of the boxes in the checklist.

**Comment by triple-underscore** [comment]
There are a number of paragraphs marked as `class="advertisement"` (and `class="deprecated"`) which was presented with a specific style (for W3C's documents). They might need an appropriate style (or need to mark up as notes, etc.).

**Comment by annevk** [comment]
@marcoscaceres see https://github.com/w3c/whatwg-coord/issues/3. That's best tracked there.

I'll have a look at the minor styling issues today. Hopefully Tab can look at the ToC one.

**Comment by annevk** [comment]
This is done. Shepherd did well it seems and you can find a Review Draft here: https://webidl.spec.whatwg.org/review-drafts/.

**Comment by ben221199** [comment]
> Just FYI, we need to figure out what to do with:
> 
> * http://www.w3.org/TR/WebIDL/ -> Redirect, I guess
> * http://www.w3.org/TR/WebIDL-1/ -> Obsolete?

I know this is an old (closed) issue, but why is http://www.w3.org/TR/WebIDL-1/ redirecting to https://webidl.spec.whatwg.org/ at this moment (28-08-2024)?

I understand that http://www.w3.org/TR/WebIDL/ does, because the one at WHATWG is the latest WebIDL specification, but I don't understand it for http://www.w3.org/TR/WebIDL-1/, because W3C published a recommendation for WebIDL Level 1 (https://www.w3.org/TR/2016/REC-WebIDL-1-20161215/) and the WHATWG specification is more like a living standard for WebIDL Level 2 (or just WebIDL 2).

I don't know if the people at WHATWG are able to disable this redirect at W3C, but if they do, I think this would make sense to do.

**Comment by annevk** [comment]
Hmm, no, the WHATWG standard supersedes all prior publications and levels. So this is accurate. This was decided in https://github.com/w3c/whatwg-coord/issues/3.

**Comment by ben221199** [comment]
I see. I agree that it should be made clear that WebIDL Level 1 is superseded and that the spec at WHATWG is the up-to-date one, but I think I don’t agree that the redirect on that specific URL should be there. As example, https://www.w3.org/TR/DOM-Level-1/ still points to the latest DOM1 spec, even when it is superseded by DOM2, DOM3, DOM4 and the living WHATWG spec DOM5.

**Comment by annevk** [comment]
Perhaps that should redirect as well. The dated copies should not redirect however.

cc @sideshowbarker

**Comment by ben221199** [comment]
No, DOM may redirect to DOM5 (the WHATWG) one, but DOM-1 should show the latest DOM-1 draft or recommendation. The same with WebIDL. WebIDL may redirect to WebIDL-2 (the WHATWG) one, but WebIDL-1 should show the latest WebIDL-1 draft or recommendation. The fact you redirect version 1 to version 2 is strange.

**Comment by annevk** [comment]
It's not your call, it's not called DOM5, and generally any of these being seen as levels or versions has been accepted as a mistake by the industry.

**Comment by ben221199** [comment]
Agreed, it isn't my call, but I just gave my opinion on why I think that redirect shouldn't be there. I called it `WebIDL 2` and `DOM5` to distinguish it from the older W3C recommendations, like we did with HTML5 (as in: the thing that came after HTML4).
